lib/Dialect/Seq/SeqOps.cpp

```diff
@@ -410,6 +415,43 @@ void CompRegClockEnabledOp::print(::mlir::OpAsmPrinter &p) {
   printCompReg(p, *this);
 }
 
+//===----------------------------------------------------------------------===//
+// ShiftRegOp
+//===----------------------------------------------------------------------===//
+
+ParseResult ShiftRegOp::parse(OpAsmParser &parser, OperationState &result) {
```

Comment:
This looks completely doable with ODS declarative format strings.

Reply:
True, if we make `parseCompReg` into a `custom` declarative format (which seems valid). AFAICT, the main driver for seq ops using custom printer/parsers is the use of `setNameFromResult`.